### DISCUSSION TITLE
How to serialize model for multiple posts?

### DISCUSSION THREAD

**Discussion by julians300** [adonisjs/core#1535] [General]
I want to serialize my post fields on a per route basis. The docs show how to do this with a single but I want them to be serialized a certain way for all. 

My current implementation is:

```js
const posts = await Post.query().where('is_published', true);
const serializedPosts = posts.map(post => post.serialize({ fields: ['title', 'author']});
```

The map seems inefficient. Is there a better way to do this?

**Comment by thetutlage** [comment]
Well, there is no way to serialize a post without looping over them. The map is not unefficient until you are looping over tens of thousands of rows.
